Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py

```diff
@@ -0,0 +1,85 @@
+# imports
+import sys
+from spellchecker import SpellChecker
+from nltk import word_tokenize
+
+# create an instance of the spellchecker
+spell = SpellChecker()
+
+# tokens --> stores the tokenized words
+tokens = []
+
+# function to read the input file
+def readTextFile(textFilename):
+    global tokens
+    words = []
+    inputFile = open(textFilename, "r")
+    tokens = word_tokenize(inputFile.read())
+
+    # Create a list of words from these tokens checking if the word is alphanumeric 
+    words = [
+        word 
+        for word in tokens if word.isalpha()
+    ]
+    inputFile.close()
+    return words
+
+# function to detect errors in the file if any
+def findErrors(textWords):
+    misspelledWords = []
+    for word in textWords:
+        # correction() --> method of spellchecker module to correct the word
+        if spell.correction(word) != word:
+            misspelledWords.append(word)
+
+    return misspelledWords
+
+# function to print errors
+def printErrors(errorList):
+    print("Misspelled words are:")
+    for word in errorList:
+        # candidates() --> method of spellchecker module to find suitable corrections of the word
+        print(f'{word} : {spell.candidates(word)}')
+
+# function to correct errors
+def correctErrors(errorList):
+    # open a new file to write the corrected text
+    outputFile = open("output.txt","w+")
+    for word in tokens:
+        if word in errorList:
+            # if word is incorrect we replace it with the corrected word
+            word = spell.correction(word)
+        
+        # this writes text to the new output.txt file
+        outputFile.write(" ".join(tokens))
+
+    outputFile.close()
+
+# main function
+def main():
```

Comment:
Add a docstring, please

And for every function remove the comment on the line before the defenition

Reply:
okay